server/fishtest/rundb.py

```diff
@@ -1163,6 +1171,48 @@ def count_games(d):
 
         return ret
 
+    def check_results(self, run, run_id, task_id):
+        old = run["results"]
```

Comment:
It seems that 'old' and 'new' refer to the same dict "results", if we are comparing the thing with itself it surely will bring up clean logs :)

Reply:
so we better repeat the testing with a copy.deepcopy() somewhere.

Reply:
get_results makes a new copy of run["results"].

Reply:
No it doesn't 

        run["results_stale"] = False
        run["results"] = results
        if save_run:
            self.buffer(run, True)

        return results
        
        you are returning a new copy but you assigned results already here to the same old reference

Reply:
```
run["results"] = results
```
The right hand side is a freshly built object.

Reply:
which we reflect on the old variable old = run["results"]
So you are comparing new vs new

Reply:
```python
old = run["results"]    # old contains reference to current results object
run["results"] = results # RHS has been built from scratch
new = run["results"]  # new contains reference to new results object
```

Reply:
Run this test program
```python
run={}
run["results"]={'a':1}
old=run["results"]
run["results"]={'a':2}
new=run["results"]
print(old,new)
```
The result is
```python
{'a': 1} {'a': 2}
```

Reply:
Okay now I understand as long as we assign the run["results"] to be a different object no reflection, but if we modify it they there will be reflection.
I'm new to python so bare with me a bit :P

Reply:
So this has nothing to do with Python. It is general computer language semantics.

Reply:
Well, yes but new is a reserved keyword ;P